src/syft/grid/services/signaling_service.py

```diff
@@ -168,21 +378,127 @@ def get_protobuf_schema() -> GeneratedProtocolMessageType:
+            msg_id=self.id.serialize(),
+            address=self.address.serialize(),
+        )
+
+    @staticmethod
+    def _proto2object(
+        proto: SignalingRequestsNotFound_PB,
+    ) -> "SignalingRequestsNotFound":
+        """Creates a SignalingAnswerMessage from a protobuf
+
+        As a requirement of all objects which inherit from Serializable,
+        this method transforms a protobuf object into an instance of this class.
+
+        :return: returns an instance of SignalingAnswerMessage
+        :rtype: SignalingAnswerMessage
+
+        .. note::
+            This method is purely an internal method. Please use syft.deserialize()
+            if you wish to deserialize an object.
+        """
+
+        return SignalingRequestsNotFound(
+            msg_id=_deserialize(blob=proto.msg_id),
+            address=_deserialize(blob=proto.address),
+        )
+
+    @staticmethod
+    def get_protobuf_schema() -> GeneratedProtocolMessageType:
+        """Return the type of protobuf object which stores a class of this type
+
+        As a part of serialization and deserialization, we need the ability to
+        lookup the protobuf object type directly from the object type. This
+        static method allows us to do this.
+
+        Importantly, this method is also used to create the reverse lookup ability within
+        the metaclass of Serializable. In the metaclass, it calls this method and then
+        it takes whatever type is returned from this method and adds an attribute to it
+        with the type of this class attached to it. See the MetaSerializable class for details.
+
+        :return: the type of protobuf object which corresponds to this class.
+        :rtype: GeneratedProtocolMessageType
+
+        """
+
+        return SignalingRequestsNotFound_PB
+
+
+class PushSignalingService(ImmediateNodeServiceWithoutReply):
     @staticmethod
     @service_auth(root_only=True)
     def process(
         node: AbstractNode,
         msg: Union[SignalingOfferMessage, SignalingAnswerMessage],
         verify_key: VerifyKey,
     ) -> None:
-        if isinstance(msg, SignalingOfferMessage):
-            print("process_offer")
-        else:
-            print("process_answer")
+        node.in_memory_client_registry[msg.id] = msg
 
     @staticmethod
-    def message_handler_types() -> List[
-        Type[Union[SignalingOfferMessage, SignalingAnswerMessage]]
-    ]:
+    def message_handler_types() -> List[Type[ImmediateSyftMessageWithoutReply]]:
         return [SignalingOfferMessage, SignalingAnswerMessage]
+
+
+class PullSignalingService(ImmediateNodeServiceWithReply):
+    @staticmethod
+    @service_auth(root_only=True)
+    def process(
+        node: AbstractNode,
+        msg: Union[OfferPullRequestMessage, AnswerPullRequestMessage],
+        verify_key: VerifyKey,
+    ) -> Union[
+        SignalingOfferMessage, SignalingAnswerMessage, SignalingRequestsNotFound
+    ]:
+        _pull_push_mapping = {
```

Comment:
`ImmediateNodeServiceWithReply.process` is a `static` method and isn't receiving `self` as a parameter. If we call `self._push_pull_mapping`, `self` will not be recognized.

Reply:
I think the suggestion Jason is making is for a class attribute which can be invoked via the ClassName.attribute rather than requiring self.attribute. I think that python adds confusion to static and class methods by letting you call self.attribute when it should be ClassName.attribute.

```python
>>> class A:
...     b = 1
...
>>> A.b
1
>>> a = A()
>>> a.b
1
```

The weird thing is, that Class Attributes are also kind of mutable singleton properties:
```python
>>> A.b = 2
>>> A.b
2
>>> a
<__main__.A object at 0x10751f940>
>>> a.b
2
```

😱

Wow it gets worse:
```python
>>> A.b = 3
>>> a.b
3
>>> a.b = 4
>>> A.b
3
```
Ah, I see whats going on, getattr fails on the instance because there is no instance attribute b, so it searches the Class and finds a class attribute. The moment we setattr on the instance we create a local attribute so future getattr resolves to that one first.

```python
>>> del a.b
>>> a.b
3
```

I started this comment with one opinion, and now I am not so sure.

So perhaps there is an argument to be made that declaring within a function provides some safer guarantees against external mutability?

Assuming we are not concerned about external modification, I have made the class attribute change and the tests pass, so that should resolve this nit.